<launch>
  <arg name="demo" default="true"/>

  <arg name="table_size"   default="[0.6, 0.8, 0.05]"/>
  <arg name="table_xyz"    default="[0.5, 0.0, 0.25]"/>
  <arg name="object_size"  default="[0.04, 0.04, 0.12]"/>
  <arg name="object_xyz"   default="[0.50, -0.10, 0.31]"/>
  <arg name="obstacle_size" default="[0.05, 0.05, 0.05]"/>
  <arg name="obstacle_xyz"  default="[0.55, 0.00, 0.35]"/>
  <arg name="place_xyz"     default="[0.50, 0.20, 0.31]"/>
  <arg name="approach_dz"   default="0.15"/>
  <arg name="lift_dz"       default="0.10"/>
  <arg name="ee_orientation" default="[0.0, 0.0, 0.0, 1.0]"/>

  <group if="$(var demo)">
    <include
      file="$(find-pkg-share franka_fer_moveit_config)/launch/demo.launch.py"/>
  </group>

  <group unless="$(var demo)">
    <node pkg="rviz2" exec="rviz2" output="log"/>
  </group>

  <node pkg="motion_planner" exec="pick_node" output="screen">
    <param name="table.size" value="$(var table_size)"/>
    <param name="table.xyz"  value="$(var table_xyz)"/>
    <param name="object.size" value="$(var object_size)"/>
    <param name="object.xyz"  value="$(var object_xyz)"/>
    <param name="obstacle.size" value="$(var obstacle_size)"/>
    <param name="obstacle.xyz"  value="$(var obstacle_xyz)"/>
    <param name="place.xyz" value="$(var place_xyz)"/>
    <param name="approach_dz" value="$(var approach_dz)"/>
    <param name="lift_dz" value="$(var lift_dz)"/>
    <param name="ee.orientation_xyzw" value="$(var ee_orientation)"/>
  </node>
</launch>
